website/blog/2022-03-03-adding-blocks-wpcli.md

```diff
@@ -0,0 +1,50 @@
+---
+title: Adding components and blocks with WP-CLI
+description: This blog post covers how to use our WP CLI commands for adding components and blocks into our projects.
+slug: adding-blocks-wpcli
+authors: iobrado
+date: 2022-03-10
+tags: [eightshift, boilerplate, wpcli, components, blocks]
+hide_table_of_contents: false
+---
+Although there are a few basic blocks available after installing the Eightshift theme, there are a lot more blocks available, but you have to add them to your project with WP-CLI. To see the complete list of available components and blocks, please visit our [Storybook](https://infinum.github.io/eightshift-docs/storybook/). 
+
+These can be used out-of-the-box, but also as a good starting point if you need similar blocks in your projects. It will also speed up your development time since you don't have to build everything from scratch.
+
+<!--truncate-->
+### Storybook
+
+Storybook allows you to preview how the components and blocks look and which options they have available. Since Storybook is interactive, you can try out most of the options. Think of it as a catalog with all custom blocks we've built and made ready for public use.
+
+Each entry in Storybook should have documentation that explains the block in more detail, along with implementation instructions. Before using any block, be sure to check `Dependencies` section. Although `Implementation` section lists all the necessary WP-CLI commands required to use a specific block or a component, it is recommended to check if you have the required dependencies ready in your theme.
+
+### WP-CLI commands
+
+If you've read our [Initial Setup](/blog/initial-setup) post, you're already familiar with our custom WP-CLI commands. For implementing additional components, we have `wp boilerplate use_component` command. For blocks, we have `wp_boilerplate use_block` command.
+
+### Adding new component and block in our project
+
+Let's say we need a Quote block in our project. After going through the documentation of Quote block, we see that we have one dependency, and that is Quote component. So, to make Quote block available in our project, we need two WP-CLI commands:
+
+```bash
+wp boilerplate use_component --name=quote
+wp boilerplate use_block --name=quote
+```
+
+The order in which you do these commands is not strict, but my personal preference is to go according to dependencies. In this case, Quote component doesn't have any dependencies, while the Quote block has one dependency, and that is Quote component. After entering these commands, run `npm start` again to make sure everything works properly.
```

Comment:
> The order in which you do these commands is not strict

It is strict - component dependencies must be resolved before a block is added, as it will throw this otherwise:
` Uncaught OverflowVendor\EightshiftLibs\Exception\InvalidBlock: Component manifest.json is missing on this location`, which will break wp-cli.

![Screenshot 2022-03-11 at 13 50 09](https://user-images.githubusercontent.com/1742806/157869773-70f30f46-7766-4a86-9244-bc56b499eef3.png)

Reply:
Oh, that's good to know, thanks for pointing that out 🙂  I've noticed in Storybook docs that the commands are written the other way around, so I thought it is not strict. I'll update that part accordingly.